libparsers/tests/csv_rw_route/preprocessor.cpp

```diff
@@ -0,0 +1,154 @@
+#include "csv_rw_route.hpp"
+#include "utils.hpp"
+#include <boost/filesystem/fstream.hpp>
+#include <doctest.h>
+
+using namespace std::string_literals;
+namespace bf = boost::filesystem;
+namespace cs = parsers::csv_rw_route;
+namespace p_util = parsers::util;
+namespace {
+
+std::string file_func(std::string const& base, std::string const& rel) {
+	return base + "/" + rel;
+}
+
+void write_to_file(std::string const& directive) {
+	bf::path p{"directive.csv"};
+	bf::ofstream ofs{p};
+	ofs << directive;
+}
+
+cs::preprocessed_lines setup(std::string const& test,
+                             parsers::errors::multi_error_t& output_errors) {
+	write_to_file(test);
+	int const seed = 1;
+	auto rng = bvereborn::datatypes::rng{seed};
+	auto processed = cs::process_include_directives("directive.csv", rng, output_errors,
+	                                                cs::file_type::csv, file_func);
+	cs::preprocess_file(processed, rng, output_errors, cs::file_type::csv);
```

Comment:
You still need to delete the file somewhere here. `boost::filesystem::delete(path)` does exactly what you want. `boost::filesystem::ostream` is basically identically to std::ostream, so it doesn't delete anything. This applies to all other places as well.